test/system/ts.mjs

```diff
@@ -42,4 +44,36 @@ tap.test('TS System Tests', async t => {
   ));
 
   t.matchSnapshot(stdout);
+
+  // The following uses a placeholder build file to simulate using
+  // PRESM with a build flag
+  const buildPath = './dist/test.mjs';
+
+  ({stdout} = await execFile(
```

Comment:
It feels like a bit of a smell that the only test requires running a subprocess. It would be better if the build logic could be called directly, e.g. as a function, without involving a new node process & loaders etc..

Reply:
Now that core logic is a bit more organized, I'll add these tests in separate PRs as discussed.